### ISSUE TITLE
BeforeRequest/AfterResponse hooks called multiple times on paginated requests

### ISSUE THREAD

**Issue by pkmnfrk** [sindresorhus/got#1367]
#### Describe the bug

 - Node.js version: v12.18.1
 - OS & version: Windows 10 1909

When using pagination features, it seems that the `beforeRequest` and `afterResponse` hooks are being called twice for each actual paged request. That is to say, the first request calls each hook twice (before, before, after, after), then the second request calls each hook twice, etc.

I have verified via Fiddler that only one actual request is being made.

#### Actual behavior
Each hook is called twice per physical request

#### Expected behavior
Each hook is only called once per physical request

#### Code to reproduce

```js
const got = require("got");

let first = true;

const client = got.extend({
    prefixUrl: "https://postman-echo.com",
    responseType: "json",
    hooks: {
        beforeRequest: [
            (request) => {
                console.log(`Before request ${request.url}`);
            }
        ],
        afterResponse: [
            (response) => {
                console.log("After response");
                return response;
            }
        ]
    },
    pagination: {
        paginate: () => {
            console.log("paginate", first);
            if(first) {
                first = false;
                return {};
            }
            return false;
        },
        countLimit: 500,
        transform: (response) => Object.keys(response.body),
    }
});

client.paginate.all("get").then((data) => {
    console.log(data);
}).catch((err) => {
    throw err;
});
```

This produces the console ouptut:
```
Before request https://postman-echo.com/get
Before request https://postman-echo.com/get
After response
After response
paginate true
Before request https://postman-echo.com/get
Before request https://postman-echo.com/get
After response
After response
paginate false
[ 'args', 'headers', 'url', 'args', 'headers', 'url' ]
```

#### Checklist

- [X] I have read the documentation.
- [X] I have tried my code with the latest version of Node.js and Got.

**Comment by hellodavie** [comment]
This issue seems to only occur when using default hooks in `got.extend()`. Here's a modified version of the code to reproduce. 

```typescript
let first = true;
let count = 0;

const gotOptions = {
  prefixUrl: "https://postman-echo.com",
  responseType: "json",
  hooks: {
    beforeRequest: [
      (request: { url: any; }) => {
        console.log(`Before request ${request.url} ${count++}`);
      }
    ],
    afterResponse: [
      (response: { url: any; }) => {
        console.log(`After response ${response.url} ${count++}`);
        return response;
      }
    ]
  },
  pagination: {
    paginate: () => {
      console.log("paginate", first);
      return false;
    },
    countLimit: 500,
    transform: (response: { body: {}; }) => Object.keys(response.body),
  }
};

function extendedClientInstance() {
  console.log('Test extendedClientInstance');
  const client = got.extend(gotOptions);
  return client.paginate.all("get");
}

function directlyCallPaginate() {
  console.log('Test directlyCallPaginate');
  return got.paginate.all('get', gotOptions);
}

function runTest(gotInstance) {
  return new Promise((resolve) => {
    first = true;
    count = 0;

    gotInstance().then((data) => {
      console.log(data);
    }).catch((err) => {
      throw err;
    }).finally(() => {
      console.log('Test Ended\n');
      resolve();
    });
  });
}

runTest(extendedClientInstance).then(() => {
  runTest(directlyCallPaginate);
});
```

The output produced is
```
Test extendedClientInstance
Before request https://postman-echo.com/get 0
Before request https://postman-echo.com/get 1
After response https://postman-echo.com/get 2
After response https://postman-echo.com/get 3
paginate true
[ 'args', 'headers', 'url' ]
Test Ended

Test directlyCallPaginate
Before request https://postman-echo.com/get 0
After response https://postman-echo.com/get 1
paginate true
[ 'args', 'headers', 'url' ]
Test Ended
```

There seems to be something wrong with how the options are normalised. Specifically,
https://github.com/sindresorhus/got/blob/61d6f610ffa6b760d466ff910a8e959778ee36e0/source/core/index.ts#L913-L914
which is set to `false`, making the hooks duplicate in the below section
https://github.com/sindresorhus/got/blob/61d6f610ffa6b760d466ff910a8e959778ee36e0/source/core/index.ts#L930-L942

I don't know if this is the best solution, but maybe set some sort of flag in the options to specify whether the instance is a paginated with custom default hooks? Then we could check for the flag and stop adding the hooks. A much simpler solution would be to check if each hook is already present before adding it, but it might not be as efficient.

**Comment by pkmnfrk** [comment]
Nice, thanks for the quick fix!

**Comment by szmarczak** [comment]
Fixed in e02845f :tada:
